c/_phonopy.c

```diff
@@ -65,6 +66,16 @@ static int distribute_fc2(double *fc2,
 			  const double t[3],
 			  const double symprec);
 
+static void distribute_fc2_with_mappings(double (*fc2)[3][3],
+                                         const int * atom_list,
+                                         const int len_atom_list,
+                                         PHPYCONST double (*r_carts)[3][3],
+                                         const int * permutations,
+                                         const int * map_atoms,
+                                         const int * map_syms,
+                                         const int num_rot,
+                                         const int num_pos);
```

Comment:
(meant to post this as a regular comment, not a review)

this function signature bothers me because map_atoms and map_syms are basically implementation details.

we could take my _get_sym_mappings_from_permutations function (at the bottom of the diff) and convert it into a C helper function, so that this function could then take atom_list_done instead of map_atoms and map_syms. I did try to do this at some point, but was discouraged by how gnarly even the simple logic in _get_sym_mappings_from_permutations becomes when written in C